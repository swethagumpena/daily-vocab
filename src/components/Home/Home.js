import React from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import styles from './Home.module.scss';

const Home = ({ vocab }) => {
  const history = useHistory();
  // const singleVocab = vocab.map((item) => (
  //   <Card
  //     word={item.addWordData.word}
  //     sentence={item.addWordData.sentence}
  //   />
  // ));
  return (
    <div>
      <p className={styles.heading}>welcome back</p>
      <div className={styles.homeContainer}>
        <div className={styles.words}>
          {vocab.map((item) => (
            <Card
              word={item.addWordData.word}
              sentence={item.addWordData.sentence}
            />
          ))}
        </div>
      </div>
      <div className={styles.AddWordBtn}>
        <button type="button" onClick={() => history.push('/add-word')}>+</button>
      </div>
    </div>
  );
};

// vocab
// [
//   { addWordData: { word: 'cogent', sentence: 'belie elish' } },
//   { addWordData: { word: 'cogent', sentence: 'belie elish' } },
//   { addWordData: { word: 'belie', sentence: 'ghjkm' } },
//   { addWordData: { word: 'abate', sentence: 'ertyu' } },
//   { addWordData: { word: 'compilant', sentence: 'belie elish' } },
// ];

// singleVocab
// [
//     {"word":"cogent","sentence":"belie elish"},
//     {"word":"cogent","sentence":"belie elish"},
//     {"word":"belie","sentence":"ghjkm"},
//     {"word":"abate","sentence":"ertyu"},
//     {"word":"compilant","sentence":"belie elish"}
// ]

const wordShape = PropTypes.shape({
  word: PropTypes.string,
  sentence: PropTypes.string,
});

const vocabShape = PropTypes.shape({
  addWordData: PropTypes.shape(wordShape),
});

Home.propTypes = {
  vocab: PropTypes.arrayOf(vocabShape).isRequired,
};

export default Home;
