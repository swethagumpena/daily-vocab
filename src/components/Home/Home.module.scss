@import "../../styles/index.scss";

.heading{
    color: $primary-text-color;
    font-family: "Inconsolata", monospace;
    font-size: 3rem;
    display: flex;
    margin: 3rem 1rem;
}

.homeContainer {
    display: flex;
    justify-content: center;
}

.words {
    display: flex;
    flex-wrap: wrap;
    padding: 1rem;
    max-width: 100rem;
    padding: 0;
}

.AddWordBtn button{
    font-family: "Inconsolata", monospace;
    padding: 1rem;
    border-radius: 50%;
    border: white;
    background-color: $primary-text-color;
    color: white;
    font-size: 1.5rem;
    margin: 2rem;
    display: flex;
}

.AddWordBtn{
    display: flex;
    justify-content: center;
}