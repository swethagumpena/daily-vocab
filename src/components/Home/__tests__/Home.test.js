/* eslint-disable react/jsx-props-no-spreading */
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import Home from '../Home';

describe(Home.name, () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const mockProps = [
    { addWordData: { word: 'abet', sentence: 'abet a suicide', id: 1 } },
  ];

  test('should match snapshot', () => {
    const { container } = render(<Home vocab={mockProps} />);
    expect(container).toMatchSnapshot();
  });

  test('should render Card component', () => {
    render(<Home vocab={mockProps} />);
    expect(screen.getByTestId('Card'));
  });

  test('should display welcome back', () => {
    render(<Home vocab={mockProps} />);
    expect(screen.getByText('welcome back'));
  });

  test('should display a + button', () => {
    render(<BrowserRouter><Home vocab={mockProps} /></BrowserRouter>);
    const addWordElement = screen.getByText('+');
    expect(addWordElement.tagName).toBe('BUTTON');
    fireEvent.click(addWordElement);
    expect(document.location.href).toBe('http://localhost/add-word');
  });
});
