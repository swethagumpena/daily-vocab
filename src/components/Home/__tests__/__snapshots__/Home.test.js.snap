// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <div
    data-testid="Home"
  >
    <p
      class="heading"
    >
      welcome back
    </p>
    <div
      class="homeContainer"
    >
      <div
        class="words"
      >
        <div
          data-testid="Card"
        >
          <div
            class="card"
          >
            <p
              class="word"
            >
              abet
            </p>
            <p
              class="sentence"
            >
              abet a suicide
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="AddWordBtn"
    >
      <button
        type="button"
      >
        +
      </button>
    </div>
  </div>
</div>
`;
